### PR TITLE
docs: add new blog about proxy Kafka by APISIX

### LINKED ISSUE #10412: help request: Can anyone provide an apisix example for the strimzi kafka
### Description

Recently, I have considered using APISIX to control Kafka authentication and authorization management, and I saw APISIX. However, I can't easily find an example of integrating APISIX with Kafka. Can anyone provide an example of how to use it? Also, It would be better to enable the go plugin with the example!

Thanks!

Note: [strimzi kafka](https://github.com/strimzi/strimzi-kafka-operator)

### Environment

- APISIX version (run `apisix version`):
- Operating system (run `uname -a`):
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`):
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`):
- APISIX Dashboard version, if relevant:
- Plugin runner version, for issues related to plugin runners:
- LuaRocks version, for installation issues (run `luarocks --version`):

### PR DESCRIPTION
**Pull request by yanmxa**
Fixes: https://github.com/apache/apisix/issues/10412

Changes:

Add the blog for proxy the Kafka cluster with apisix

Screenshots of the change:

### CODE REVIEW

**Comment by pottekkat** [comment]
Thanks for the contribution @yanmxa!

File: blog/en/blog/2023/12/26/access-kafka-by-apisix.md [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+  - name: Meng Yan
+    title: Author
+    url: https://github.com/yanmxa
+keywords:
+  - Message Queue
+  - Kafka
+  - Gateway
+  - Strimzi
+description: >
+  A few days ago, I tried to add a proxy to the kafka cluster, so I can leverage the proxy to manage the kafka authentication and authorization. Next, I will show how to use the APISIX to develop a customize authorization plugin for the kafka cluster.
```
Comment (yzeng25) [inline blog/en/blog/2023/12/26/access-kafka-by-apisix.md]:
```suggestion
description: A few days ago, I tried to add a proxy to the kafka cluster, so I can leverage the proxy to manage the kafka authentication and authorization. Next, I will show how to use the APISIX to develop a customize authorization plugin for the kafka cluster.
```

File: blog/en/blog/2023/12/26/access-kafka-by-apisix.md [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+curl -X GET 'http://localhost:9080/consumers/strimzi-kafka-consumer-group/instances/strimzi-kafka-consumer/records' \
+  -H 'Host: example.com' \
+  -H 'Accept: application/vnd.kafka.json.v2+json' \
+  -H 'Source: client' \
+  -H "Client-Certificate: $CERT_CONTENT_BASE64" \
+```
+
+## Reference
+
+- https://apisix.apache.org/zh/docs/ingress-controller/tutorials/how-to-use-go-plugin-runner-in-apisix-ingress/
```
Comment (yzeng25) [inline blog/en/blog/2023/12/26/access-kafka-by-apisix.md]:
```suggestion
- [How to use go-plugin-runner with APISIX Ingress](https://apisix.apache.org/zh/docs/ingress-controller/tutorials/how-to-use-go-plugin-runner-in-apisix-ingress/)
```

File: blog/en/blog/2023/12/26/access-kafka-by-apisix.md [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+- https://www.apiseven.com/blog/go-makes-apache-apisix-better
```
Comment (yzeng25) [inline blog/en/blog/2023/12/26/access-kafka-by-apisix.md]:
```suggestion
- [How to use Go to develop Apache APISIX plugin](https://apisix.apache.org/blog/2021/08/19/go-makes-apache-apisix-better/)
```

File: blog/en/blog/2023/12/26/access-kafka-by-apisix.md [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+- https://zhuanlan.zhihu.com/p/613540331
```
Comment (yzeng25) [inline blog/en/blog/2023/12/26/access-kafka-by-apisix.md]:
```suggestion
- [APISIX之Go插件开发](https://zhuanlan.zhihu.com/p/613540331)
```

File: blog/en/blog/2023/12/26/access-kafka-by-apisix.md [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+- https://www.fdevops.com/2022/10/09/casbin-apisix-31182
```
Comment (yzeng25) [inline blog/en/blog/2023/12/26/access-kafka-by-apisix.md]:
```suggestion
- [结合 casbin 为 APISIX 开发一个接口权限校验插件](https://www.fdevops.com/2022/10/09/casbin-apisix-31182)
```

File: blog/en/blog/2023/12/26/access-kafka-by-apisix.md [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+- https://blog.csdn.net/weixin_42873928/article/details/123279381
```
Comment (yzeng25) [inline blog/en/blog/2023/12/26/access-kafka-by-apisix.md]:
```suggestion
- [Docker部署 apisix 并使用golang插件(自定义鉴权方式)](https://blog.csdn.net/weixin_42873928/article/details/123279381)
```

File: blog/en/blog/2023/12/26/access-kafka-by-apisix.md [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+  - Message Queue
+  - Kafka
+  - Gateway
+  - Strimzi
+description: >
+  A few days ago, I tried to add a proxy to the kafka cluster, so I can leverage the proxy to manage the kafka authentication and authorization. Next, I will show how to use the APISIX to develop a customize authorization plugin for the kafka cluster.
+
+tags: [Ecosystem]
+---
+
```
Comment (yzeng25) [inline blog/en/blog/2023/12/26/access-kafka-by-apisix.md:16]:
```suggestion
---

> This blog shows how to use Apache APISIX to develop a customize authorization plugin for the kafka cluster.

<!--truncate-->

```

**Comment by yanmxa** [comment]
Thanks @yzeng25 !
Done!

**Comment by yzeng25** [comment]
@pottekkat PTAL at the contents.

File: blog/en/blog/2023/12/26/access-kafka-by-apisix.md [hunk 8]
```diff
@@ -0,0 +1,272 @@
...
+  --namespace apisix
+```
+
+- Configure the Kafka Route with Admin API
+
+```bash
+# forward 9180 port to local host
+kubectl -n apisix port-forward $(kubectl get pods -l app.kubernetes.io/name=apisix -n apisix -o jsonpath="{.items[0].metadata.name}") 9180:9180
+
+# the bridge service name can be accessed by
```
Comment (yzeng25) [inline blog/en/blog/2023/12/26/access-kafka-by-apisix.md]: fix lint, remove trailing space.
```suggestion
# the bridge service name can be accessed by
```

File: blog/en/blog/2023/12/26/access-kafka-by-apisix.md [hunk 9]
```diff
@@ -0,0 +1,272 @@
...
+...
+# Nginx will hide all environment variables by default. So you need to declare your variable first in the conf/config.yaml
+# https://github.com/apache/apisix/blob/master/docs/en/latest/external-plugin.md
+nginx_config:
+  envs:
+    - APISIX_LISTEN_ADDRESS
+    - APISIX_CONF_EXPIRE_TIME
+
+ext-plugin:
+  # path_for_test: "/tmp/runner.sock"
```
Comment (yzeng25) [inline blog/en/blog/2023/12/26/access-kafka-by-apisix.md]: fix lint, remove trailing space.
```suggestion
  # path_for_test: "/tmp/runner.sock"
```

File: blog/en/blog/2023/12/26/access-kafka-by-apisix.md [hunk 10]
```diff
@@ -0,0 +1,272 @@
...
+CERT_CONTENT_BASE64=$(base64 < rest/client.crt)
+
+# list the topics
+curl -i 'http://127.0.0.1:9080/topics' \
+-H 'Host: example.com' \
+-H 'Content-Type: application/vnd.kafka.json.v2+json' \
+-H 'Source: client' \
+-H "Client-Certificate: $CERT_CONTENT_BASE64"
+
+# create consumer
```
Comment (yzeng25) [inline blog/en/blog/2023/12/26/access-kafka-by-apisix.md]: fix lint, remove trailing space.
```suggestion
# create consumer
```

File: blog/en/blog/2023/12/26/access-kafka-by-apisix.md [hunk 11]
```diff
@@ -0,0 +1,272 @@
...
+
+# consume message
+curl -X GET 'http://localhost:9080/consumers/strimzi-kafka-consumer-group/instances/strimzi-kafka-consumer/records' \
+  -H 'Host: example.com' \
+  -H 'Accept: application/vnd.kafka.json.v2+json' \
+  -H 'Source: client' \
+  -H "Client-Certificate: $CERT_CONTENT_BASE64" \
+```
+
+## Reference
```
Comment (yzeng25) [inline blog/en/blog/2023/12/26/access-kafka-by-apisix.md]:
```suggestion
## References
```

File: blog/en/blog/2023/12/26/access-kafka-by-apisix.md [hunk 12]
```diff
@@ -0,0 +1,272 @@
...
+
+- [How to use go-plugin-runner with APISIX Ingress](https://apisix.apache.org/zh/docs/ingress-controller/tutorials/how-to-use-go-plugin-runner-in-apisix-ingress/)
+- [How to use Go to develop Apache APISIX plugin](https://apisix.apache.org/blog/2021/08/19/go-makes-apache-apisix-better/)
+- [APISIX之Go插件开发](https://zhuanlan.zhihu.com/p/613540331)
+- [结合 casbin 为 APISIX 开发一个接口权限校验插件](https://www.fdevops.com/2022/10/09/casbin-apisix-31182)
+- [Docker部署 apisix 并使用golang插件(自定义鉴权方式)](https://blog.csdn.net/weixin_42873928/article/details/123279381)
```
Comment (yzeng25) [inline blog/en/blog/2023/12/26/access-kafka-by-apisix.md]: fix lint, add a new line eof.
```suggestion
- [Docker部署 apisix 并使用golang插件(自定义鉴权方式)](https://blog.csdn.net/weixin_42873928/article/details/123279381)

```

**Review by yzeng25 — DISMISSED** [review]

File: blog/en/blog/2023/12/26/access-kafka-by-apisix.md [hunk 13]
```diff
@@ -0,0 +1,272 @@
...
+    title: Author
+    url: https://github.com/yanmxa
+keywords:
+  - Message Queue
+  - Kafka
+  - Gateway
+  - Strimzi
+description: A few days ago, I tried to add a proxy to the kafka cluster, so I can leverage the proxy to manage the kafka authentication and authorization. Next, I will show how to use the APISIX to develop a customize authorization plugin for the kafka cluster.
+
+tags: [Ecosystem]
```
Comment (yzeng25) [inline blog/en/blog/2023/12/26/access-kafka-by-apisix.md]:
```suggestion
description: A few days ago, I tried to add a proxy to the kafka cluster, so I can leverage the proxy to manage the kafka authentication and authorization. Next, I will show how to use the APISIX to develop a customize authorization plugin for the kafka cluster.
tags: [Ecosystem]
```

File: blog/en/blog/2023/12/26/access-kafka-by-apisix.md [hunk 14]
```diff
@@ -0,0 +1,271 @@
...
+    title: Author
+    url: https://github.com/yanmxa
+keywords:
+  - Message Queue
+  - Kafka
+  - Gateway
+  - Strimzi
+description: A few days ago, I tried to add a proxy to the kafka cluster, so I can leverage the proxy to manage the kafka authentication and authorization. Next, I will show how to use the APISIX to develop a customize authorization plugin for the kafka cluster.
+tags: [Ecosystem]
+---
```
Comment (yzeng25) [inline blog/en/blog/2023/12/26/access-kafka-by-apisix.md:15]: @Yilialinn @JYan00 will you add a cover image now or later?

Comment (saltism) [reply blog/en/blog/2023/12/26/access-kafka-by-apisix.md:15]: https://static.apiseven.com/uploads/2023/12/29/oJyQz1c4_A&K_Cover.png

**Review by saltism — CHANGES_REQUESTED** [review]

File: blog/en/blog/2023/12/26/access-kafka-by-apisix.md [hunk 15]
```diff
@@ -0,0 +1,271 @@
...
+
+> This blog shows how to use Apache APISIX to develop a customize authorization plugin for the kafka cluster.
+
+<!--truncate-->
+
+## Prerequisites
+
+- Openshift cluster
```
Comment (kayx23) [inline blog/en/blog/2023/12/26/access-kafka-by-apisix.md]:
```suggestion
- Have a running OpenShift cluster
```

File: blog/en/blog/2023/12/26/access-kafka-by-apisix.md [hunk 16]
```diff
@@ -0,0 +1,271 @@
...
+- Kafka cluster created by [strimzi kafka operator](https://github.com/strimzi/strimzi-kafka-operator)
```
Comment (kayx23) [inline blog/en/blog/2023/12/26/access-kafka-by-apisix.md]:
```suggestion
- Run a Kafka cluster with [strimzi kafka operator](https://github.com/strimzi/strimzi-kafka-operator)
```

File: blog/en/blog/2023/12/26/access-kafka-by-apisix.md [hunk 17]
```diff
@@ -0,0 +1,271 @@
...
+- kubectl, [oc](https://docs.openshift.com/container-platform/4.11/cli_reference/openshift_cli/getting-started-cli.html) and curl
```
Comment (kayx23) [inline blog/en/blog/2023/12/26/access-kafka-by-apisix.md]:
```suggestion
- Install [kubectl](https://kubernetes.io/docs/reference/kubectl), [OpenShift CLI](https://docs.openshift.com/container-platform/4.11/cli_reference/openshift_cli/getting-started-cli.html) and curl on host
```

File: blog/en/blog/2023/12/26/access-kafka-by-apisix.md [hunk 18]
```diff
@@ -0,0 +1,271 @@
...
+
+## Expose the Kafka Cluster by KafkaBridge
+
+To simplify the configuration setting for the kafka. I provision the kafka by [strimzi-kafka-operator](https://github.com/strimzi/strimzi-kafka-operator). In order to make Kafka expose interfaces externally like other services, I use `KafkaBridge` to transform it into an HTTP service.
+
+- Create the `KafkaBridge`
```
Comment (kayx23) [inline blog/en/blog/2023/12/26/access-kafka-by-apisix.md:31]: A general suggestion: no need to use bullet points for each of the steps; you could write full sentences.

File: blog/en/blog/2023/12/26/access-kafka-by-apisix.md [hunk 19]
```diff
@@ -0,0 +1,271 @@
...
+authors:
+  - name: Meng Yan
+    title: Author
+    url: https://github.com/yanmxa
+keywords:
+  - Message Queue
+  - Kafka
+  - Gateway
+  - Strimzi
+description: A few days ago, I tried to add a proxy to the kafka cluster, so I can leverage the proxy to manage the kafka authentication and authorization. Next, I will show how to use the APISIX to develop a customize authorization plugin for the kafka cluster.
```
Comment (pottekkat) [inline blog/en/blog/2023/12/26/access-kafka-by-apisix.md]:
```suggestion
description: A few days ago, I added Apache APISIX as a proxy to an Apache Kafka cluster to manage authentication and authorization. Now, I created a custom authorization plugin in Go for the Kafka cluster.
```

File: blog/en/blog/2023/12/26/access-kafka-by-apisix.md (hunk 16 above)
Comment (pottekkat) [inline blog/en/blog/2023/12/26/access-kafka-by-apisix.md]:
```suggestion
- Kafka cluster created by [strimzi-kafka-operator](https://github.com/strimzi/strimzi-kafka-operator)
```

File: blog/en/blog/2023/12/26/access-kafka-by-apisix.md [hunk 20]
```diff
@@ -0,0 +1,271 @@
...
+    title: Author
+    url: https://github.com/yanmxa
+keywords:
+  - Message Queue
+  - Kafka
+  - Gateway
+  - Strimzi
+description: A few days ago, I added Apache APISIX as a proxy to an Apache Kafka cluster to manage authentication and authorization. Now, I created a custom authorization plugin in Go for the Kafka cluster.
+tags: [Ecosystem]
+---
```
Comment (yzeng25) [inline blog/en/blog/2023/12/26/access-kafka-by-apisix.md]:
```suggestion
tags: [Ecosystem]
image: https://static.apiseven.com/uploads/2023/12/29/oJyQz1c4_A&K_Cover.png
---
```

**Review by yzeng25 — APPROVED** [review]

**Review by moonming — APPROVED** [review]

**Pull request merged**